src/librustdoc/json/conversions.rs

```diff
@@ -130,11 +140,11 @@ impl FromWithTcx<clean::GenericArgs> for GenericArgs {
         use clean::GenericArgs::*;
         match args {
             AngleBracketed { args, bindings } => GenericArgs::AngleBracketed {
-                args: args.into_vec().into_iter().map(|a| a.into_tcx(tcx)).collect(),
-                bindings: bindings.into_iter().map(|a| a.into_tcx(tcx)).collect(),
+                args: args.into_vec().into_tcx(tcx),
```

Comment:
It seems to be doing the same thing but instead of creating one vector, it creates two. Did I miss something?

Reply:
Ah no nevermind. But then I wonder why the `into_vec` is needed at all...

Reply:
Probably it could skip the `into_vec` because of the impl for IntoIterator right?

Reply:
Like @camelid said. :)

Reply:
`args` is a `Box<[GenericArg]>`, which doesn't impl `IntoIterator`, but `Vec` does and the conversion is zero cost. I guess you could `impl <T, U: FromWithTcx<T>> FromWithTcx<Box<[T]>> for Vec<U>`, but I don't think thats worth it as it only comes up twice.

Reply:
Huh strange. Well if it doesn't work don't worry about it then.